### PR TITLE
140 new resource securityroles 1

### LINKED ISSUE #140: [New Resource]: /security/roles
### Description

/security/roles 

### Requested Resource(s) and/or Data Source(s)

* netapp-ontap_security_roles

### Potential Terraform Configuration

_No response_

### References

https://docs.netapp.com/us-en/ontap-restapi/ontap/swagger-ui/index.html#/security/role_create

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by wenjun666**

### CODE REVIEW

**Review by carchi8py — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ FEATURES:
 * **New Data Source:** `netapp-ontap_volumes_files` ([#8](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/8))
 * **New Data Source:** `netapp-ontap_quota_rules` ([#135](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/135))
 * **New Data Source:** `netapp-ontap_quota_rule` ([#135](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/135))
+* **New Data Source:** `netapp-ontap_security_role` ([#139](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/139))
+* **New Data Source:** `netapp-ontap_security_roles` ([#139](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/139))
 * **New Data Source:** `netapp-ontap_security_login_message` ([#17](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/17))
```
Comment (chuyich) [inline CHANGELOG.md]: Missing the resource of security_role

File: docs/resources/security_role.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+- `scope` (String) Scope of the entity. Set to 'cluster' for cluster owned objects and to 'svm' for SVM owned objects.
+
+<a id="nestedatt--privileges"></a>
+### Nested Schema for `privileges`
+
+Optional:
+
+- `access` (String) Access level for the REST endpoint or command/command directory path. If it denotes the access level for a command/command directory path, the only supported enum values are 'none','readonly' and 'all'.
+- `path` (String) Either of REST URI/endpoint OR command/command directory path.
+- `query` (String) Optional attribute that can be specified only if the 'path' attribute refers to a command/command directory path. The privilege tuple implicitly defines a set of objects the role can or cannot access at the specified access level. The query further reduces this set of objects to a subset of objects that the role is allowed to access. The query attribute must be applicable to the command/command directory specified by the 'path' attribute. It is defined using one or more parameters of the command/command directory path specified by the 'path' attribute.
```
Comment (chuyich) [inline docs/resources/security_role.md]: Missing the import part

**Review by chuyich — CHANGES_REQUESTED** [review]

File: docs/resources/security_role.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    {
+    access = "all"
+    path = "lun"
+  },
+  	{
+	  access = "all"
+	  path = "vserver"
+	  query = "-vserver acc_test"
+	}
+  ]
```
Comment (suhasbshekar) [inline docs/resources/security_role.md]: intendation

File: examples/resources/netapp-ontap_security_role/resource.tf [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  privileges = [
+  {
+    access = "all"
+    path = "lun"
+  },
+  {
+	  access = "all"
+	  path = "vserver"
+	  query = "-vserver acc_test"
+	}
```
Comment (suhasbshekar) [inline examples/resources/netapp-ontap_security_role/resource.tf]: indentation

**Review by suhasbshekar — CHANGES_REQUESTED** [review]

File: docs/resources/security_role.md (hunk 2 above)
Comment (wenjun666) [reply docs/resources/security_role.md]: what about the import part?

File: docs/resources/security_role.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "netapp-ontap_security_role Resource - terraform-provider-netapp-ontap"
+subcategory: ""
```
Comment (chuyich) [inline docs/resources/security_role.md]: Missing subcatetory

File: docs/resources/security_role.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+Create/Modify/Delete a SecurityRole resource
+
+### Related ONTAP commands
+```commandline
+* security login role create
+* security login role show
+* security login role delete
+* security login role modify
+```
+
```
Comment (chuyich) [inline docs/resources/security_role.md]: Missing supported platforms

**Review by chuyich — CHANGES_REQUESTED** [review]
Besides the commented parts. Maybe add the change based on agreed resource naming? Or do it later is ok, too.

**Review by suhasbshekar — APPROVED** [review]

**Review by chuyich — COMMENTED** [review]
Wondered if query is part of the privilege in this resource. Please double check it.

**Comment by wenjun666** [comment]
> Wondered if query is part of the privilege in this resource. Please double check it.

I don't think it matters, as query is privilege.query, if API returns nothing of query, the struct will omit empty.

**Review by chuyich — APPROVED** [review]

**Pull request merged**